File: backend/models/states.js
/**
 * Mental state enum and index lookups.
 * Port of backend/models/states.py
 */

export const MentalState = {
  ANXIETY: 'ACUTE ANXIETY (Visada)',
  DEPRESSION: 'DEPRESSION / LETHARGY (Tamas)',
  STRESS: 'STRESS & AGITATION (Krodha)',
  EQUILIBRIUM: 'EQUILIBRIUM (Sattva)',
};

export const STATE_LABELS = Object.values(MentalState);

export const STATE_INDEX = {};
STATE_LABELS.forEach((label, i) => {
  STATE_INDEX[label] = i;
});

export const STATE_KEYS = Object.keys(MentalState);

File: backend/models/states.py
"""Mental state enum and index lookups."""

from __future__ import annotations

from enum import Enum
from typing import Dict, List


class MentalState(str, Enum):
    ANXIETY     = "ACUTE ANXIETY (Visada)"
    DEPRESSION  = "DEPRESSION / LETHARGY (Tamas)"
    STRESS      = "STRESS & AGITATION (Krodha)"
    EQUILIBRIUM = "EQUILIBRIUM (Sattva)"


STATE_LABELS: List[str] = [s.value for s in MentalState]

STATE_INDEX: Dict[str, int] = {s.value: i for i, s in enumerate(MentalState)}
